plugins/scaffolder-field-extensions/scaffolder-frontend-module-http-request-field/src/components/SelectFieldFromApi/SelectFieldFromApi.tsx

```diff
@@ -114,6 +114,17 @@ const SelectFieldFromApiComponent = (
       })}?${params}`,
       { headers },
     );
+    if (options.successStatusCode) {
+      if (response.status !== options.successStatusCode) {
+        throw new Error(
+          `Failed to retrieve data from API: ${response.statusText}`,
+        );
+      }
+    } else if (!response.ok) {
```

Comment:
Maybe by default it should default to the existing behavior and we can make this minor, thoughts?